lib/interval/interval.ex

```diff
@@ -220,8 +249,14 @@ defmodule Timex.Interval do
       if has_recursion_ended?(current_date, end_date, right_open) do
         {:done, acc}
       else
-        next_date = Timex.shift(current_date, keywords)
-        do_reduce({next_date, end_date, right_open, keywords}, fun.(current_date, acc), fun)
+        case Timex.shift(current_date, keywords) do
+          {:error, {:unknown_shift_unit, :invalid_step}} ->
```

Comment:
We shouldn't use specific step value here; we'd better use ```{:error, {:unknown_shift_unit, _}} ->``` instead.
You could check it by using another step value in specs:

```elixir
interval = %Interval{from: ~D[2017-04-02], until: ~D[2017-05-02], step: [another: 1]}
assert_raise Interval.FormatError, "Invalid step unit for %Timex.Interval{}", fn ->
  Enum.count(interval)
end
```
It will fail.